tests/unit/package_managers/pip/test_rust.py

```diff
@@ -0,0 +1,29 @@
+from pathlib import Path
+
+import pytest
+
+from hermeto.core.package_managers.pip.rust import _infer_rust_root_dir_from
+
+
+@pytest.mark.parametrize(
+    "cargo_files,expected_rust_root_dir",
+    [
+        ((Path("/tmp/foo/Cargo.toml"), Path("/tmp/bar/baz/Cargo.toml")), Path("/tmp/foo")),
+        ((Path("/tmp/bar/baz/Cargo.toml"), Path("/tmp/foo/Cargo.toml")), Path("/tmp/foo")),
+        (
+            (
+                Path("/tmp/bar/baz/Cargo.toml"),
+                Path("/tmp/foo/Cargo.toml"),
+                Path("/tmp/foo/quux/Cargo.toml"),
+            ),
+            Path("/tmp/foo"),
+        ),
+    ],
+)
+def test_the_shortest_path_in_cargo_package_is_inferred_as_root(
```

Comment:
What is supposed to happen if we'd encounter a package that would have `<root>/src/pkg1/Cargo.toml` and `<root>/src/pkg2/Cargo.toml`? The inference currently uses `min` , but since `glob` is unstable so is our inference in that case. Sorting would only help stabilizing the tests, but it'd do nothing in real life. I guess that's a topic for a different time though, this PR is only simplifying existing code, it was just a thought from the top of my head.

Reply:
if it comes to that, I think we would have to check `pyproject.toml` configuration.